wolvic/browser/vr/wvr_manager.cc

```diff
@@ -850,4 +875,62 @@ void WvrManager::UpdateLayerBounds(int16_t frame_index,
   CreateOrResizeWebXrSurface(source_size);
 }
 
+device::mojom::XREnvironmentBlendMode
+WvrManager::PickEnvironmentBlendModeForSession(
+    device::mojom::XRSessionMode session_mode) {
+  DCHECK(wvr_api_->get_system_state().enumerationCompleted);
+
+  const auto& display_state = wvr_api_->get_system_state().displayState;
+  auto supportsBlendMode =
+      [&display_state](const mozilla::gfx::VRDisplayBlendMode blend_mode) {
+        for (auto& supported_blend_mode : display_state.blendModes) {
+          if (supported_blend_mode ==
```

Comment:
We are doing this check every time we call the supportsBlendMode labmda, but the display_state is the same, isn't ?  isn't going to be the same result each time ?

Reply:
We only call `PickEnvironmentBlendMode` from `WvrManager::ConnectPresentingService` which means that it's called just once for every WebXR experience that is started. We could indeed cache it but it's so cheap that I thought it wouldn't pay off.

Reply:
The `_empty` value marks the end of the array, is the end condition so you could not ignore it. The vector has a fixed size (because it's part of a shared memory area) so we need to mark the end of it.